trappy/ftrace.py

```diff
@@ -216,15 +204,42 @@ def _update_cache(self):
             warnings.warn(
                 "TRAPpy: Cache not created due to OS error: {0}".format(err))
 
-    def _load_cache(self):
-        params = self._get_params_to_cache()
+    def _max_window(self):
+        """
+        Compute the maximum window of the 'window' & 'abs_window' intersection
+        """
+        max_window = [0, None]
+        max_window[0] = max(self.window[0] + self.basetime, self.abs_window[0])
```

Comment:
This is because of how windows are handled in the parsing (and is one of the reasons why I really don't like how windows are handled in TRAPpy, I thought you were right but it took me a while to re-understand what's going on) in `__populate_data`:

```
if (timestamp < self.window[0] + self.basetime) or \
    (timestamp < self.abs_window[0]):
    self.lines += 1
    continue
```

In other words, an event is skipped if its timestamp is lower than *either* bound. This means events will start being considered if their timestamp is greater than the greatest lower bound, so `max` should fit the bill.

Reply:
Probably never happens because we don't use windows most of the time, and if we do it's either `window` or `abs_window`, not both.